### PR TITLE
Look at oldest send operation, rather than number of running ones

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review. 

Operations are put in a collection, but only with leases from a semaphore. 
The number of operations in this collection is checked to see whether anything happens — if the number of running operations is the same over a given timeout, `close()` short-cuts and doesn't wait. 

The change here is to check the operation ID of the oldest running operation instead.

### CODE REVIEW

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/operationProcessor/OperationProcessor.java [hunk 1]
```diff
@@ -103,6 +106,15 @@ public int getIncompleteResultQueueSize() {
         }
     }
 
+    /** Returns the id of the oldest operation to be sent. */
+    public Optional<String> oldestIncompleteResultId() {
+        synchronized (monitor) {
+            return Optional.of(docSendInfoByOperationId.keySet().iterator())
```
Comment (bratseth) [inline vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/operationProcessor/OperationProcessor.java]: Why can't you just say return 
docSendInfoByOperationId.isEmpty() ? Optional.empty() : Optional.of(docSendInfoByOperationId.keySet().iterator().next)
like every normal person?  :-)

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/api/FeedClientImpl.java [hunk 2]
```diff
@@ -60,14 +61,13 @@ public void stream(String documentId, CharSequence documentData, Object context)
     @Override
     public void close() {
         Instant lastResultReceived = Instant.now();
```
Comment (bratseth) [inline vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/api/FeedClientImpl.java]: Strictly this should be called something else now.

**Review by bratseth — APPROVED** [review]
It looks fine, but what's wrong with the old method?

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/operationProcessor/OperationProcessor.java (hunk 1 above)
Comment (jonmv) [reply vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/operationProcessor/OperationProcessor.java]: Hehe. You don't have a bad point.

File: vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/api/FeedClientImpl.java (hunk 2 above)
Comment (jonmv) [reply vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/api/FeedClientImpl.java]: Yes :(

**Comment by bratseth** [comment]
Ah, nice catch!

**Pull request merged**